osu.Game.Rulesets.Osu/Utils/OsuHitObjectGenerationUtils.cs

```diff
@@ -100,5 +104,53 @@ public static Vector2 RotateVectorTowardsVector(Vector2 initial, Vector2 destina
                 initial.Length * MathF.Sin(finalAngleRad)
             );
         }
+
+        /// <summary>
+        /// Reflects an OsuHitObject's position horizontally.
+        /// </summary>
+        /// <param name="osuObject">The OsuHitObject to be reflected.</param>
+        /// <returns>The reflected OsuHitObject.</returns>
+        public static OsuHitObject ReflectOsuHitObjectHorizontally(OsuHitObject osuObject)
```

Comment:
I would shorten it to `ReflectHitObjectHorizontally` or even just `ReflectHorizontally` (same for vertical version).

Also this shouldn't return the `OsuHitObject` if it's the same reference, feels like it would return a clone that way.